src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/ExprToolRuleSpeed.java

```diff
@@ -0,0 +1,90 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.components.ToolComponent.ToolRule;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.toolcomponent.ToolRuleWrapper;
+import org.skriptlang.skript.bukkit.toolcomponent.elements.EffSecCreateToolRule.ToolRuleEvent;
+
+@Name("Tool Rule - Speed")
+@Description("The speed of a tool rule determining the mining speed of the blocks of the tool rule for an item.")
+@Examples({
+	"create a new tool rule and store it in {_toolrule}:",
+		"\tset the tool rule blocks to oak log, stone and obsidian",
+		"\tset the tool rule speed to 5",
+		"\tenable tool rule drops",
+	"add {_toolrule} to the tool rules of {_item}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class ExprToolRuleSpeed extends PropertyExpression<ToolRule, Number> {
```

Comment:
I wonder if it's worth using `Float` instead of `Number` here, I believe skriptlang has said in the past expected types should reflect the return type only time it shouldn't is in syntax that should be kept as `%number%`

Reply:
I'm not entirely sure, see what the team says,

Reply:
if this always returns a float, you should make it a float